pkg/backend/display/detailedDiff.go

```diff
@@ -0,0 +1,220 @@
+package display
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi/pkg/engine"
+	"github.com/pulumi/pulumi/pkg/resource"
+	"github.com/pulumi/pulumi/pkg/resource/plugin"
+	"github.com/pulumi/pulumi/pkg/util/contract"
+)
+
+func parseDiffPath(path string) ([]interface{}, error) {
+	// Complete paths obey the following EBNF-ish grammar:
+	//
+	//   propertyName := [a-zA-Z_$] { [a-zA-Z0-9_$] }
+	//   quotedPropertyName := '"' ( '\' '"' | [^"] ) { ( '\' '"' | [^"] ) } '"'
+	//   arrayIndex := { [0-9] }
+	//
+	//   propertyIndex := '[' ( quotedPropertyName | arrayIndex ) ']'
+	//   rootProperty := ( propertyName | propertyIndex )
+	//   propertyAccessor := ( ( '.' propertyName ) |  propertyIndex )
+	//   path := rootProperty { propertyAccessor }
+	//
+	// We interpret this a little loosely in order to keep things simple. Specifically, we will accept something close
+	// to the following:
+	// pathElement := { '.' } ( '[' ( [0-9]+ | '"' ('\' '"' | [^"] )+ '"' ']' | [a-zA-Z_$][a-zA-Z0-9_$] )
+	// path := { pathElement }
+
+	var elements []interface{}
+	for len(path) > 0 {
+		switch path[0] {
+		case '.':
+			path = path[1:]
+		case '[':
+			// If the character following the '[' is a '"', parse a string key.
+			var pathElement interface{}
+			if path[1] == '"' {
+				var propertyKey []byte
+				var i int
+				for i = 2; ; {
+					if i == len(path) {
+						return nil, errors.New("missing closing quote in property name")
+					} else if path[i] == '"' {
+						i++
+						break
+					} else if path[i] == '\\' && i+1 < len(path) && path[i+1] == '"' {
```

Comment:
Just checking here, there's no way to escape backslashes (i.e. do we have to transform `\\` to `\`?)

Reply:
We only treat `\` as an escape character if it is followed by a `"`, so there should be no need to escape backslashes.

Reply:
Gotcha - For my own memory if I ever come back to read this, that means that to get the literal string `\"` you would write `\\"`.  I agree that this code behaves as you desire.